websites/Y/YouTube Music/presence.ts

```diff
@@ -277,8 +283,10 @@ function updateSongTimestamps(useTimeLeft: boolean) {
 			.split(" / "),
 		[currTimes, totalTimes] = element;
 
-	mediaTimestamps = presence.getTimestamps(
-		useTimeLeft ? presence.timestampFromFormat(currTimes) : Date.now(),
-		useTimeLeft ? presence.timestampFromFormat(totalTimes) : null
-	);
+	if (useTimeLeft) {
+		mediaTimestamps = presence.getTimestamps(
+			presence.timestampFromFormat(currTimes),
+			presence.timestampFromFormat(totalTimes)
+		);
+	} else mediaTimestamps[0] = Math.floor(Date.now() / 1000);
```

Comment:
Start timestamp will be "reset" each time `updateSongTimestamps` is called.
I think we want to show the elapsed time from the playing song instead.

Reply:
I'm pretty sure setting the timestamp to now does nothing (it's just a theory bc it works fine for me)

Reply:
Setting the timestamp to `Date.now()` will reset the elapsed time. There's no time difference between now and now.
So each time you call it, you will reset it. which happens when a user unpauses the song or seeks.
If the user is listening to a song at 3.34 / 4.10 we want to show 3m34s elapsed, equivalent to showing 36s left.

Reply:
yer I know but it....does that. I'm confused now. Can you clone it and see if it works for you or just some strange bug for me?

Reply:
Right now you are setting Date.now to the start timestamp each time `updateSongTimestamps` is called.
I don't need to run the presence locally to see that it doesn't show the actual elapsed time of the song.

Reply:
but it....it does. Like it works perfectly fine for me

Reply:
Don't "it works on my machine" me now 😭

Reply:
HDJEJDIEJDIR IT DOES

Reply:
BUT ITS NOT THE BEHAVIOR WE WANT GOOD SIR

Reply:
OKOK IN A BIT I WILL GO OVER IT FJENIDNEIDBEIDE

Reply:
So take date.now and subtract the elapsed song time from it